frontend/src/metabase/dashboard/actions/auto-wire-parameters/utils.ts

```diff
@@ -0,0 +1,133 @@
+import type { Draft } from "@reduxjs/toolkit";
+import _ from "underscore";
+import type {
+  CardId,
+  DashboardCard,
+  DashboardId,
+  DashCardId,
+  ParameterId,
+  ParameterTarget,
+} from "metabase-types/api";
+import type { DashboardState } from "metabase-types/store";
+import { isActionDashCard } from "metabase/actions/utils";
+import { getExistingDashCards } from "metabase/dashboard/actions/utils";
+import { isVirtualDashCard } from "metabase/dashboard/utils";
+import { getParameterMappingOptions } from "metabase/parameters/utils/mapping-options";
+import { compareMappingOptionTargets } from "metabase-lib/parameters/utils/targets";
+import type Metadata from "metabase-lib/metadata/Metadata";
+
+export function getAllDashboardCardsWithUnmappedParameters(
+  dashboard: Draft<DashboardState>,
+  dashboardId: DashboardId,
+  parameter_id: ParameterId,
+  excludeDashcardIds: DashCardId[] = [],
+) {
+  return getExistingDashCards(dashboard, dashboardId).filter(dashcard => {
+    return (
+      !excludeDashcardIds.includes(dashcard.id) &&
+      !dashcard.parameter_mappings?.some(
+        mapping => mapping.parameter_id === parameter_id,
+      )
+    );
+  });
+}
+
+export function getMatchingParameterOption(
+  dashcardToCheck: DashboardCard,
+  targetDimension: ParameterTarget,
+  targetDashcard: DashboardCard,
+  metadata: Metadata,
+) {
+  if (!dashcardToCheck) {
+    return [];
+  }
+
+  return getParameterMappingOptions(
+    metadata,
+    null,
+    dashcardToCheck.card,
+    // TODO: mapping-options.js/getParameterMappingOptions needs to be converted to typescript as the TS
```

Comment:
Not really sure how to approach this one. Since `mapping-options.js` is JS and the 4th parameter is `dashcard = null`, typescript thinks the parameter must be `null` or `undefined`. I think it'll just need to keep this unless someone has a better solution